.github/workflows/amplify-deploy.yml

```diff
@@ -107,13 +107,43 @@ jobs:
             --app-id ${{ inputs.amplify-app-id }} \
             --branch-name ${{ env.BRANCH_NAME }} \
             --source-url https://${{ env.BUCKET_NAME }}.s3.${{ env.AWS_REGION }}.amazonaws.com/${{ env.ZIP_NAME }}
+      # Set env vars for domain name association
+      - name: Set environment variables for domain name association
+        if: github.event_name == 'pull_request'
+        run: |
+          PACKAGE_NAME_SUFFIX=$(echo "${{ inputs.package-name }}" | sed -e 's/^pie-//')
+          echo "SUB_DOMAIN=pr${{ github.event.number }}-$PACKAGE_NAME_SUFFIX" >> $GITHUB_ENV
+
+
+      - name: get existing domain associations
```

Comment:
This code gets the existing domain associations via the aws CLI.

It then uses JQ to put the existing associations in the following format:

**prefix** - the sub-domain
**branchName** - The name of the branch in amplify.

- The output then gets stored in an environment variable called `DOMAIN_ASSOCIATIONS`, which is then passed to the following step when we call the `aws amplify update-domain-association` command

`[{prefix: "pr<PR_NUMBER>-<APP-NAME>", "branchName: pr<PR_NUMBER>"}`

- If no association is present, a new entry gets added.